hazelcast-client/src/main/java/com/hazelcast/client/spi/impl/ClientTransactionManagerServiceImpl.java

```diff
@@ -106,33 +111,66 @@ public String getGroupName() {
     }
 
     public ClientConnection connect() throws Exception {
```

Comment:
A method named `connect` that returns `Clientconnection` looks like it should be in `ConnectionManager`, isn't it?

Reply:
Logic is only needed to this service though. I am not strongly against that, but I am mostly avoiding put the new logic in central services (like ConnectionManager), if  no one else is going to use it. 
If we move this to ClientConnectionManager, this would be named `getOrConnectToRandomConnection` 
In that case, I would probably make `timeout` and `retryPause` parameterized in order not to mix  connectionManager and invocationService.